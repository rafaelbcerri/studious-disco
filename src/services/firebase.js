export const login = (email, password) => (
  firebase
    .auth()
    .signInWithEmailAndPassword(email, password)
)