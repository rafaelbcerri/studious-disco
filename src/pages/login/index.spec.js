import * as Firebase from '../../services/firebase.js';
import { Login } from './index.js';

describe('Login', () => {
  beforeEach(() => {
    Firebase.login = jest.fn(() => Promise.resolve(true));
  })

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('should be a function', () => {
    expect(typeof Login).toBe('function');
  });

  it('should render the page', () => {
    expect(Login()).toMatchSnapshot();
  })

  it('should submit login with email and password', () => {
    const loginEl = Login();

    loginEl.querySelector('#email').value = 'email@example.com';
    loginEl.querySelector('#password').value = '123123';

    const loginForm = loginEl.querySelector('#login-form');
    loginForm.dispatchEvent(new Event('submit'));

    expect(Firebase.login).toHaveBeenCalledWith("email@example.com", "123123");
  })
});
