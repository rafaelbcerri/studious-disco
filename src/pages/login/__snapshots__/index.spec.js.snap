// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the page 1`] = `
<div>
  
    
  <div>
    
      
    <h1>
      Login
    </h1>
    
      
    <form
      id="login-form"
    >
      
        
      <input
        id="email"
        type="email"
      />
      
        
      <input
        id="password"
        type="password"
      />
      
        
      <button>
         Entrar 
      </button>
      
      
    </form>
    
    
  </div>
  
  
</div>
`;
